lib/elixir/pages/Writing Documentation.md

```diff
@@ -48,6 +48,38 @@ The compiler will infer this argument as `map`. Sometimes the inference will be
 
     def size(map)
 
+## Functions with multiple clauses
+
+When documenting a function with multiple clauses, it is recommended to add the documentation
+and typespecs to the function head rather than adding @doc and @spec to each clause of
+the function.
+
+#### Example
+
+    @doc """
+    Adds tax to the amount and returns the amount inclusive of tax rounded to 2 decimal places.
+
+    ## Example
+
+        add_tax(10.2, 0.23)
+        add_tax(10.2, country: "Ireland")
+    """
+    @spec add_tax(amount :: float, tax_param :: integer | float | Keyword.t) :: float
+    def add_tax(amount, tax_param \\ 0)
+
+    def add_tax(amount, tax_param) when is_number(tax_param) do
+      total = amount + calc_tax(amount, tax_param)
+      Float.round(total, 2)
+    end
+
+    def add_tax(amount, country: country) do
+      total = amount + calc_tax(amount, get_tax_rate(country))
+      Float.round(total, 2)
+    end
+
+    defp calc_tax(amount, rate), do: amount * rate
+    defp get_tax_rate("Ireland"), do: 0.23
```

Comment:
Do you think we could reduce the complexity of this function? I'm afraid it overshadows what we're trying to showcase here :). We could use whatever simple function that requires a pattern match.